---
title: "Class 16: Pathway analysis from RNA-seq results"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Differential Expression Analysis

My input files for todays class
```{r}
metaFile <- "data/GSE37704_metadata.csv"
countFile <- "data/GSE37704_featurecounts.csv"
```

Read the metadata file first
```{r}
#Import metadata and take a peak
colData = read.csv(metaFile, row.names=1)
head(colData)
```

```{r}
# Import countdata
countData = read.csv(countFile, row.names=1)
head(countData)
```

We need the countData and colData files to match up so we will need to remove that odd first column in countData namely contData$length.
```{r}
# Note we need to remove the odd first $length col
countData <- as.matrix(countData[,-1])
head(countData)
```

```{r}
dim(countData)
```

check now that metadata columns and rows match to coldata
```{r}
all(rownames(colData) == colnames(countData))
```

Complete the code below to filter countData to exclude genes (i.e. rows) where we have 0 read count across all samples 
```{r}
# Filter count data where you have 0 read count across all samples.
#countData = countData[ ,]
#head(countData)

#experiment
#head(countData)
countData = countData[rowSums(countData) != 0,]
head(countData)
```

>How many genes do we have left?

```{r}
nrow(countData)
```


##Running DESeq2
gear, add "deseq_load" and check off "show messages"
```{r deseq_load, message=FALSE}
library(DESeq2)
```

design=~condition, as condition is a column in metadata file that we want to compare
```{r}
dds = DESeqDataSetFromMatrix(countData=countData,
                             colData=colData,
                             design=~condition)
dds = DESeq(dds)
```

Lets get out results
```{r}
res = results(dds, contrast=c("condition", "hoxa1_kd", "control_sirna"))
```

>Q. Call the summary() function on your results to get a sense of how many genes are up or down-regulated at the default 0.1 p-value cutoff.

```{r}
summary(res)
```

```{r}
plot(res$log2FoldChange, -log(res$padj))
```

```{r}
head(res)
```

>Q. Improve this plot by completing the below code, which adds color and axis labels

```{r}
# Make a color vector for all genes. this makes everything gray.
mycols <- rep("gray", nrow(res) )

# Color red the genes with absolute fold change above 2 and P < 0.01. this makes the extremities red.
mycols[ abs(res$log2FoldChange) > 2 ] <- "red"

plot(res$log2FoldChange, -log(res$padj), col=mycols)

# Color blue those with adjusted p-value less than 0.01
#  and absolute fold change more than 2
inds <- (abs(res$log2FoldChange) > 2) & (res$padj < 0.01)
mycols[ inds ] <- "blue"

plot( res$log2FoldChange, -log(res$padj), col= mycols, xlab="Log2(FoldChange)", ylab="-Log(P-value)" )
```

```{r}
library("AnnotationDbi")
library("org.Hs.eg.db")
columns(org.Hs.eg.db)
```

```{r}
res$symbol = mapIds(org.Hs.eg.db,
                    keys=row.names(res), 
                    keytype="ENSEMBL",
                    column="SYMBOL",
                    multiVals="first")
res$entrez = mapIds(org.Hs.eg.db,
                    keys=row.names(res),
                    keytype="ENSEMBL",
                    column="ENTREZID",
                    multiVals="first")
res$name =   mapIds(org.Hs.eg.db,
                    keys=row.names(res),
                    keytype="ENSEMBL",
                    column="GENENAME",
                    multiVals="first")
head(res,10)
```

>Q. Finally for this section letâ€™s reorder these results by adjusted p-value and save them to a CSV file in your current project directory.

```{r}
res = res[order(res$pvalue),]
write.csv(res,file="deseq_results.csv")
```

##Section 2 Pathway Analysis
KEGG pathways

```{r eval=FALSE}
biocLite( c("pathview", "gage", "gageData") )
```

```{r}
library(pathview)
```

```{r}
library(gage)
library(gageData)
```

```{r}
data(kegg.sets.hs)
data(sigmet.idx.hs)

# Focus on signaling and metabolic pathways only
kegg.sets.hs = kegg.sets.hs[sigmet.idx.hs]
# Examine the first 3 pathways
head(kegg.sets.hs, 3)
```

```{r}
foldchanges = res$log2FoldChange
names(foldchanges) = res$entrez
head(foldchanges)
```

run the gage pathway analysis
```{r}
# Get the results
keggres = gage(foldchanges, gsets=kegg.sets.hs)
```

```{r}
#Now lets look at the object returned from gage().
attributes(keggres)
```

```{r}
# Look at the first few down (less) pathways
head(keggres$less)
```

```{r}
pathview(gene.data=foldchanges, pathway.id="hsa04110")
```
![](hsa04110.pathview.png)
```{r}
# A different PDF based output of the same data
pathview(gene.data=foldchanges, pathway.id="hsa04110", kegg.native=FALSE)
```

## Focus on top 5 upregulated pathways here for demo purposes only

```{r}
keggrespathways <- rownames(keggres$greater)[1:5]
# Extract the 8 character long IDs part of each string
keggresids = substr(keggrespathways, start=1, stop=8)
keggresids
```

```{r}
pathview(gene.data=foldchanges, pathway.id=keggresids, species="hsa")
```
